### ISSUE TITLE
Cannot link to a non running container

### ISSUE THREAD

**Issue by kokosing** [prestodb/presto-admin#351]
https://api.travis-ci.org/jobs/303016970/log.txt?deansi=true

```

======================================================================
ERROR: test_connection_to_coord_lost (tests.product.test_server_install.TestServerInstall)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/prestodb/presto-admin/tests/product/test_server_install.py", line 384, in test_connection_to_coord_lost
    coordinator=down_node, pa_raise_error=False)
  File "/home/travis/build/prestodb/presto-admin/tests/product/standalone/presto_installer.py", line 52, in install
    rpm_name = self.copy_presto_rpm_to_master(cluster=cluster)
  File "/home/travis/build/prestodb/presto-admin/tests/product/standalone/presto_installer.py", line 98, in copy_presto_rpm_to_master
    if not self._check_rpm_already_uploaded(self.rpm_name, cluster):
  File "/home/travis/build/prestodb/presto-admin/tests/product/standalone/presto_installer.py", line 144, in _check_rpm_already_uploaded
    'ls ' + os.path.join(cluster.rpm_cache_dir, rpm_name)
  File "/home/travis/build/prestodb/presto-admin/tests/docker_cluster.py", line 292, in exec_cmd_on_host
    user=user)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/docker/utils/decorators.py", line 34, in wrapper
    return f(self, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/docker/api/exec_api.py", line 71, in exec_create
    return self._result(res, True)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/docker/api/client.py", line 226, in _result
    self._raise_for_status(response)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/docker/api/client.py", line 222, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
APIError: 500 Server Error: Internal Server Error ("Cannot link to a non running container: /slave1-988f65a7-656f-4fd2-b92b-92b5f8351829 AS /master-7e44a3ed-c948-4c8e-8cc5-23d4b921b2a5/slave1-988f65a7-656f-4fd2-b92b-92b5f8351829")
```

**Comment by kokosing** [comment]
Above stack trace is from #349 PR testing, the below occurred to me in internal automation (jenkins).

```
======================================================================
ERROR: test_connection_to_coord_lost (tests.product.test_server_install.TestServerInstall)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/tests/product/test_server_install.py", line 384, in test_connection_to_coord_lost
    coordinator=down_node, pa_raise_error=False)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/tests/product/standalone/presto_installer.py", line 52, in install
    rpm_name = self.copy_presto_rpm_to_master(cluster=cluster)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/tests/product/standalone/presto_installer.py", line 98, in copy_presto_rpm_to_master
    if not self._check_rpm_already_uploaded(self.rpm_name, cluster):
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/tests/product/standalone/presto_installer.py", line 144, in _check_rpm_already_uploaded
    'ls ' + os.path.join(cluster.rpm_cache_dir, rpm_name)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/tests/docker_cluster.py", line 320, in exec_cmd_on_host
    tty=tty, user=user)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/venv/local/lib/python2.7/site-packages/docker/utils/decorators.py", line 35, in wrapper
    return f(self, *args, **kwargs)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/venv/local/lib/python2.7/site-packages/docker/utils/decorators.py", line 21, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/venv/local/lib/python2.7/site-packages/docker/api/exec_api.py", line 38, in exec_create
    return self._result(res, True)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/venv/local/lib/python2.7/site-packages/docker/client.py", line 146, in _result
    self._raise_for_status(response)
  File "/tmp/jenkins-5e6c9ce2/workspace/presto-admin-build-pipeline/venv/local/lib/python2.7/site-packages/docker/client.py", line 142, in _raise_for_status
    raise errors.APIError(e, response, explanation=explanation)
APIError: 500 Server Error: Internal Server Error ("Cannot link to a non running container: /slave1-bf84a689-d7d9-4392-9165-8c6a5906ebcd AS /master-08c8bfcc-e5c4-4de3-af0f-9d34f750c563/slave1-bf84a689-d7d9-4392-9165-8c6a5906ebcd")
```

CC @petroav Do you recall anything similar?

**Comment by kokosing** [comment]
Fixed with #353
